pkg/messages/baseMessage.go

```diff
@@ -63,12 +68,17 @@ type ResponseMessage interface {
 type BaseMessage struct {
 	senderFloor int
 	messageType MessageType
+	id          string
 }
 
-func NewBaseMessage(senderFloor int, messageType MessageType) *BaseMessage {
+func NewBaseMessage(senderFloor int, messageType MessageType, id string) *BaseMessage {
```

Comment:
Isn't there a way to make this id optional and it fills itself to empty by default? (instead of having all new messages forced to add empty ID)

Reply:
Yes there is. And also... not how I would do this anyway

Reply:
Nevermind lol, Go does not support optional arguments nor function overloading... I guess this is the way...

Reply:
@JaafarRammal  @hussain2603  you can use variadic arguments https://stackoverflow.com/a/19813113